### PR TITLE
Add slice operation to direct bindings

### PR DESCRIPTION
**Pull request by rdspring1**

### CODE REVIEW

**Comment by rdspring1** [comment]
!test

**Comment by rdspring1** [comment]
!build

File: tests/python/opinfo/opinfos.py [hunk 1]
```diff
@@ -1265,6 +1265,7 @@ def torch_reshape_sym_fn(input_tensor, output_shaped_tensor):
     error_input_generator=slice_error_generator,
     reference=jax.lax.slice if JAX_AVAILABLE else None,
     reference_type=ReferenceType.Jax,
+    supports_direct_bindings=True,
```
Comment (jjsjann123) [inline tests/python/opinfo/opinfos.py:1268]: Neat~

File: tests/python/direct/test_python_frontend.py [hunk 2]
```diff
@@ -774,3 +774,36 @@ def fusion_func(fd: FusionDefinition):
...
+    def fusion_func(fd: FusionDefinition) -> None:
+        T0 = fd.define_tensor(
+            shape=[-1, -1, -1],
+            contiguity=[True, True, True],
+            dtype=DataType.Float,
+            is_cpu=False,
+            stride_order=[2, 1, 0],
+        )
+        T1 = fd.ops.slice(
+            T0, start_indices=offset, end_indices=(2, 5, 10), strides=(1, 1, 1)
```
Comment (jjsjann123) [inline tests/python/direct/test_python_frontend.py:793]: nitpick: felt a bit strange that we have `offset` but not `(2, 5, 10)` as a named variable.

File: python/python_direct/python_translate.cpp [hunk 3]
```diff
@@ -904,6 +904,41 @@ class PythonTranslator : public OptInConstDispatch {
...
+        "start_indices", "end_indices", "strides", "manual_normalization"};
+    printer_.generateKwargsOperation(
+        "fd.ops.slice",
+        std::make_tuple(sop->in()),
+        slice_argument_names,
+        std::make_tuple(
+            start_indices,
+            stop_indices,
+            strides,
+            /*manual_normalization=*/true),
```
Comment (jjsjann123) [inline python/python_direct/python_translate.cpp:941]: shouldn't we use the variant with default value? since the python API has strides and `manual_normalization` with default value?

Comment (jjsjann123) [reply python/python_direct/python_translate.cpp:941]: ah, I see the point is that normalization from direct binding should have already be expressed in the fusion IR. So manual_normalization should always be true.

Can we add a comment on that if that's the case?

**Review by jjsjann123 — APPROVED** [review]
LGTM

File: python/python_direct/python_translate.cpp (hunk 3 above)
Comment (rdspring1) [reply python/python_direct/python_translate.cpp:941]: Yes, I thought it would be very difficult to pattern match on the manual normalization, so it would always be true.

File: tests/python/direct/test_python_frontend.py (hunk 2 above)
Comment (rdspring1) [reply tests/python/direct/test_python_frontend.py:793]: `offset` is used for `V_start`.

**Comment by rdspring1** [comment]
!build
